sympy/integrals/transforms.py

```diff
@@ -1169,6 +1734,59 @@ def _collapse_extra(self, extra):
                 'Laplace', None, 'No combined convergence.')
         return plane, cond
 
+    def doit(self, **hints):
+        """
+        Try to evaluate the Laplace transform in closed form.
+        """
+
+        needeval = hints.pop('needeval', False)
```

Comment:
Rather than using `pop` it would be better to use explicit keyword only arguments:
```python
def doit(self, *, needeval=False, **hints):
```

Reply:
I believe you, but this is also from the original:

https://github.com/sympy/sympy/blob/b5191b805eba30488a54cc0e46dbab66052a0429/sympy/integrals/transforms.py#L151-L153

There would then be three keywords that would need to be made explicit. Should we change this in both locations?

Reply:
The original code probably predates explicit keyword-only argments:
https://www.python.org/dev/peps/pep-3102/

> There would then be three keywords that would need to be made explicit.

Do they all need to be made explicit? The point is just that mutating data structures like this is fragile whereas using an explicit parameter for `needeval` is less bug-prone.

Reply:
I see the point, but I suggest the following: I will re-write `_laplace_apply_rules` such that it uses a keryword-only argument, but will not touch the needeval/doit part. For that I suggest to open a new "code cleanup" issue and solve it for all integral transforms independent of the functionality change I am doing here.

Reply:
I think really the problem is that the `doit` method from the superclass should not be duplicated here.

Reply:
The lines

https://github.com/sympy/sympy/blob/b5191b805eba30488a54cc0e46dbab66052a0429/sympy/integrals/transforms.py#L152-L153

in the superclass hurt me badly.  I also need `AppliedUndef` to be deferred to my `_compute_transform`; this is why duplicated `doit`. Is there a possibility to change the `doit` in the superclass such that it behaves differently when used in the `LaplaceTranform`?

Reply:
Yes, just move that code into a separate method and override the method in the subclass.

Reply:
Thank you for teaching me proper object-oriented programming here!

(I guess you must already found out that OOP is not something I have actually learnt properly and that I am making things up as I go along ;)

Understood, I will do this!

Reply:
No problem. The existing code here is quite messy. We need to do our best not to make that worse... (when possible)